dev/tasks/tasks.yml

```diff
@@ -1013,6 +1013,15 @@ tasks:
   test-r-devdocs:
     ci: github
     template: r/github.devdocs.yml
+    
+  test-r-depsource-auto:
+      ci: azure
+      template: r/azure.linux.yml
+      params:
+        r_org: rocker
+        r_image: r-base
+        r_tag: latest
+        flags: '-e ARROW_DEPENDENCY_SOURCE=AUTO'
```

Comment:
Should we try and add a task that also tests `ARROW_DEPENDENCY_SOURCE=SYSTEM`? To test when we can't fall back to building?  Maybe I'm paranoid, but I'm a bit worried if the rocker base happens to not have any of our dependencies installed, this isn't _actually_ exercising linking to them.

Reply:
We can also make a follow up for this, I don't think that `ARROW_DEPENDENCY_SOURCE=SYSTEM` will work with this image without extending it a bit to install the other dependencies we need.